p2p/src/authenticated/actors/peer/actor.rs

```diff
@@ -124,55 +136,29 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng> Actor<E> {
                                 Some(msg_control) => msg_control,
                                 None => return Err(Error::PeerDisconnected),
                             };
-                            match msg {
-                                Message::BitVec { bit_vec } => {
-                                    let msg = wire::Message{
-                                        payload: Some(wire::message::Payload::BitVec(bit_vec)),
-                                    }.encode_to_vec();
-                                    conn_sender.send(&msg)
-                                        .await
-                                        .map_err(Error::SendFailed)?;
-                                    self.sent_messages
-                                        .get_or_create(&metrics::Message::new_bit_vec(&peer))
-                                        .inc();
-                                }
-                                Message::Peers { peers: msg } => {
-                                    let msg = wire::Message{
-                                        payload: Some(wire::message::Payload::Peers(msg)),
-                                    }.encode_to_vec();
-                                    conn_sender.send(&msg)
-                                        .await
-                                        .map_err(Error::SendFailed)?;
-                                    self.sent_messages
-                                        .get_or_create(&metrics::Message::new_peers(&peer))
-                                        .inc();
-                                }
+                            let (payload, metric_msg) = match msg {
+                                Message::BitVec { bit_vec } =>
+                                    (Payload::BitVec(bit_vec), metrics::Message::new_bit_vec(&peer)),
+                                Message::Peers { peers: msg } =>
+                                    (Payload::Peers(msg), metrics::Message::new_peers(&peer)),
                                 Message::Kill => {
                                     return Err(Error::PeerKilled(hex(&peer)))
                                 }
-                            }
+                            };
+                            Self::send(&mut conn_sender,payload,metric_msg,&self.sent_messages)
```

Comment:
nit: the spacing on this function call isn't correct (I think the linter will catch this but let's see).